### PR TITLE
#7565: Set layer visibility limits from Catalog

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR adds a new advanced settings option to WMS service of Catalog to be configured. With this option, user can now configure to services to automatically include visibility limits info to the layer which can later be viewed and modified from layer settings in TOC
Option: **Auto-set Visibility Limits**

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature
 - [x] Refactoring

## Issue

**What is the current behavior?**
#7565 

**What is the new behavior?**
- Configure WMS service to automatically include visibility limits info on adding layer from catalog
- Configuration can be done directly from advanced settings of WMS service in Catalog or from localConfig

**Example WMS service cfg from localConfig**
```
catalog: {
  services: {
    "service_wms": {
      "url": "https://gs-stable.geo-solutions.it/geoserver/wms",
      "type": "wms",
      "title": "GeoSolutions GeoServer WMS",
      "autoSetVisibilityLimits": true
    }
  }
}
```

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
Ref: https://github.com/georchestra/mapstore2-georchestra/issues/424

### CODE REVIEW

File: docs/developer-guide/local-config.md [hunk 1]
```diff
@@ -176,6 +176,7 @@ Set `selectedService` value to one of the ID of the services object ("Demo CSW S
   "readOnly": "if true, makes the service not editable from catalog plugin"
   "titleMsgId": "optional, string used to localize the title of the service, the string must be present in translations",
   "format": "image/png8" // the image format to use by default for layers coming from this catalog (or tiles).
+  "autoSetVisibilityLimits": "if true, allows to fetch and set visibility limits of the layer from capabilities"
```
Comment (offtherailz) [inline docs/developer-guide/local-config.md:179]:
```suggestion
  "autoSetVisibilityLimits": "if true, allows to fetch and set visibility limits of the layer from capabilities on layer add"
```

File: web/client/translations/data.it-IT.json [hunk 2]
```diff
@@ -1437,6 +1437,10 @@
                     "helpText": "Usa il segnaposto ${searchText} nel modello per inserire la stringa di ricerca"
                 },
                 "error": "Sintassi xml non valida"
+            },
+            "autoSetVisibilityLimits": {
+                "label": "Imposta limite di visibilità",
+                "tooltip": "Imposta automaticamente i limiti di visibilità del livello dalle capacità"
```
Comment (offtherailz) [inline web/client/translations/data.it-IT.json:1443]:
```suggestion
                "label": "Imposta limite di visibilità",
                "tooltip": "Applica automaticamente i limiti di visibilità suggeriti dal server"
```

File: web/client/translations/data.en-US.json [hunk 3]
```diff
@@ -1436,7 +1436,11 @@
                     "helpText": "Use template with ${searchText} placeholder to capture search string"
                 },
                 "error": "Invalid xml syntax"
-            }
+            },
+           "autoSetVisibilityLimits": {
+                "label": "Set Visibility Limit",
+                "tooltip": "Automatically set visibility limits of the layer from capabilities"
```
Comment (offtherailz) [inline web/client/translations/data.en-US.json:1442]:
```suggestion
                "tooltip": "Automatically applies the visibility limits suggested by the server"
```

File: web/client/translations/data.fr-FR.json [hunk 4]
```diff
@@ -1437,6 +1437,10 @@
                     "helpText": "Utilisez un modèle avec un espace réservé ${searchText} pour capturer la chaîne de recherche"
                 },
                 "error": "Syntaxe xml non valide"
+            },
+            "autoSetVisibilityLimits": {
+                "label": "Définir la limite de visibilité",
+                "tooltip": "Définir automatiquement les limites de visibilité de la couche à partir des capacités"
```
Comment (offtherailz) [inline web/client/translations/data.fr-FR.json:1443]:
```suggestion
                "tooltip": "Applique automatiquement les limites de visibilité suggérées par le serveur"
```

File: web/client/translations/data.es-ES.json [hunk 5]
```diff
@@ -1436,6 +1436,10 @@
                     "helpText": "Utilice la plantilla con el marcador de posición $ {searchText} para capturar la cadena de búsqueda"
                 },
                 "error": "Sintaxis XML no válida"
+            },
+            "autoSetVisibilityLimits": {
+                "label": "Establecer límite de visibilidad",
+                "tooltip": "Establezca automáticamente los límites de visibilidad de la capa a partir de las capacidades."
```
Comment (offtherailz) [inline web/client/translations/data.es-ES.json:1442]:
```suggestion
                "tooltip": "Aplica automáticamente los límites de visibilidad sugeridos por el servidor"
```

**Review by offtherailz — CHANGES_REQUESTED** [review]

File: web/client/translations/data.de-DE.json [hunk 6]
```diff
@@ -1474,6 +1474,10 @@
                     "helpText": "Verwenden Sie eine Vorlage mit dem Platzhalter ${searchText}, um die Suchzeichenfolge zu erfassen"
                 },
               "error": "Ungültige XML-Syntax"
+            },
+            "autoSetVisibilityLimits": {
+                "label": "Sichtbarkeitslimit festlegen",
+                "tooltip": "Legen Sie automatisch Sichtbarkeitsgrenzen des Layers aus den Funktionen fest"
```
Comment (offtherailz) [inline web/client/translations/data.de-DE.json:1480]:
```suggestion
                "tooltip": "Wendet automatisch die vom Server vorgeschlagenen Sichtbarkeitsgrenzen an"
```

**Review by offtherailz — APPROVED** [review]

**Comment by offtherailz** [comment]
@ElenaGallo, could you please test this on  DEV ? Thank you

**Comment by ElenaGallo** [comment]
Test passed, @dsuren1 please backport to 2021.02.xx

**Pull request merged**